### ISSUE TITLE
escape special char from get parameters

### ISSUE THREAD

**Issue by coim32** [opengeospatial/ets-wcs20#45]
on test wcs2:get-kvp-core-req28
Format parameter in line 715 of script/ctl/GetCoverage.xmlif  can read format "application/gml+xml" and tries to send it as a get parameter it will fail because the character '+', the parameter value should be escaped in this case it should send "application/gml%2Bxml"

**Comment by dstenger** [comment]
@coim32 Can you provide a test service to reproduce the error?

**Comment by keshav-nangare** [comment]
@coim32 

Used service for testing: http://moon.planetserver.eu:8084/rasdaman/ows?service=WCS&request=GetCapabilities&acceptversions=2.0.1

The wcs2:get-kvp-core-req28 is successfully executed even if it contains the + sign in query parameter.

E.g.:  http://moon.planetserver.eu:8084/rasdaman/ows?service=WCS&VERSION=2.0.1&request=GetCoverage&coverageid=test_mr&format=application/gml+xml

Also I have tested with the POSTMAN rest client and it is executed successfully. The issue is not reproducible.

Is there any specific case or you can provide the steps to reproduce it?

**Comment by coim32** [comment]
That parameter is not properly scaped, our implementation is not released yet, otherwise I could give you a link to test it, you can see in this link the correct value is encoded and decoded https://urldecode.org/?text=application%252fgml%252bxml&mode=decode

Does postman and rasdaman works with the encoded value?
--------------------------------------------
On Thu, 12/28/17, Keshav Nangare <notifications@github.com> wrote:

 Subject: Re: [opengeospatial/ets-wcs20] escape special char from get parameters (#45)
 To: "opengeospatial/ets-wcs20" <ets-wcs20@noreply.github.com>
 Cc: "coim32" <luisarc0@yahoo.com>, "Mention" <mention@noreply.github.com>
 Date: Thursday, December 28, 2017, 6:45 AM
 
 @coim32
 Used service for testing: http://moon.planetserver.eu:8084/rasdaman/ows?service=WCS&request=GetCapabilities&acceptversions=2.0.1
 The wcs2:get-kvp-core-req28 is successfully executed even
 if it contains the + sign in query parameter.
 E.g.:  http://moon.planetserver.eu:8084/rasdaman/ows?service=WCS&VERSION=2.0.1&request=GetCoverage&coverageid=test_mr&format=application/gml+xml
 Also I have tested with the POSTMAN rest client and it is
 executed successfully. The issue is not reproducible.
 Is there any specific case or you can provide the steps
 to reproduce it?
 
 —
 You are receiving this because you were mentioned.
 Reply to this email directly, view
 it on GitHub, or mute
 the thread.

**Comment by coim32** [comment]
My implementation is not released yet, encoding is part of http specification to scape special characters.
You can see it working in this link https://urldecode.org/?text=application%252fgml%252bxml&mode=decode
--------------------------------------------
On Wed, 12/20/17, dstenger <notifications@github.com> wrote:

 Subject: Re: [opengeospatial/ets-wcs20] scape special char from get parameters (#45)
 To: "opengeospatial/ets-wcs20" <ets-wcs20@noreply.github.com>
 Cc: "coim32" <luisarc0@yahoo.com>, "Mention" <mention@noreply.github.com>
 Date: Wednesday, December 20, 2017, 2:05 AM
 
 @coim32 Can you
 provide a test service to reproduce the error?
 
 —
 You are receiving this because you were mentioned.
 Reply to this email directly, view
 it on GitHub, or mute
 the thread.

**Comment by keshav-nangare** [comment]
Updated the wcs20 test with the encoding function which will encode the request parameter.

The encoding issue of the test `wcs2:get-kvp-core-req28` is fixed.

```
Test wcs2:get-kvp-core-req28 type Mandatory default result Passed (s0253/d9730e37_1/d9730e225_1/d1e214_1/d8198e149_1)

Assertion: Req28: If a GetCoverage request contains a format parameter then this parameter shall contain a MIME type identifier occurring in some wcs:formatSupported element of the response to a successful GetCapabilities request to this server.

Request d8198e1216_1:
   Method: get
   URL: http://moon.planetserver.eu:8084/rasdaman/ows?service=WCS&VERSION=2.0.1&request=GetCoverage&coverageid=test_mr&format=application%2Fgml%2Bxml
   Response from parser parsers:HTTPParser:
      <response>
   <status code="200" protocol="HTTP/1.1">OK</status>
   <headers>
      <header name="Server">Apache-Coyote/1.1</header>
      <header name="File-name">test_mr.gml</header>
      <header name="Content-disposition">inline; filename=test_mr.gml</header>
      <header name="Content-Type">text/xml</header>
      <header name="Transfer-Encoding">chunked</header>
      <header name="Date">Thu, 18 Jan 2018 08:30:55 GMT</header>
   </headers>
```

**Comment by keshav-nangare** [comment]
Fixed with this commit: 50775f6620cbf077f00cebbec7046176a44f871a. [PR: #53]

**Comment by dstenger** [comment]
Successfully tested.

Test wcs2:get-kvp-core-req28 is passed.
